labs/lab6/lab6.adoc

```diff
@@ -29,56 +37,311 @@ Password: INSTRUCTOR WILL PROVIDE
+
+== Dynamic Scaleup Architecture
+
+The _0-Self-Configure_ job template executed earlier configured a set of templates to automate the process to expand the OpenShift cluster. A workflow job template called **2-Scaleup_OpenShift_on_AWS** was also created as a result. The workflow first creates a new AWS instance and once the instance has been created, the scaleup Ansible playbook will be executed to expand the cluster. Finally, a third job template performs final actions to finalize the cluster scaleup.
+
+As part of the deployment of the Prometheus within OpenShift, another component of the Prometheus ecosystem includes a notification component called link:https://prometheus.io/docs/alerting/alertmanager/[AlertManager]. Alerts can be sent to a variety of endpoints ranging from commercial offerings such as Slack, HipChat and PagerDuty, or to generic endpoints using webhooks. 
 
-== Review Cluster
+As mentioned earlier, Ansible Tower features a Rest API for querying resources, including triggering Workflow/Job Templates. 
 
-Recall the number of nodes in the cluster by either visiting CloudForms or Ansible Tower.
 
-From the Ansible Tower:
+Unfortunately, the API is not compatible with the webhook notifications being emitted by AlertManager. To support the webhook function of AlertManager to trigger Job templates in Ansible Tower, a new "bridge" application will be developed and deployed to OpenShift.  
 
-* Click on **INVENTORIES** in the top navigation pane
-* Click on **OpenShift** inventory
-* Click on the **Groups** button to view the groups within the inventory
-* Select the **OSEv3** group and review child groups and the instances within the _Hosts_ section
+== Create an Ansible Playbook Bundle ==
 
-The lists of instances in the OpenShift cluster can also be viewed from the OpenShift master using the OpenShift CLI:
+In link:../lab5/lab5.adoc[Lab 5], an Ansible Playbook Bundle was used to deploy MediaWiki which was backed by a PostgreSQL database. In this lab, an Ansible Playbook Bundle will be created to build and deploy the AlertManager/Tower bridge application.
+
+Developing Ansible Playbook bundles requires the following:
+
+* The Ansible Playbook Bundle CLI Tool (apb)
+* A user logged in to OpenShift
+* Permissions to access cluster level resources
+
+The Ansible Playbook Bundle CLI (apb) is available in the OpenShift yum repositories which have already been enabled on the OpenShift master. Execute the following command to install the _apb_ CLI:
 
 .master$
 [source, bash]
 ----
-oc get nodes
+yum install -y apb
```

Comment:
Just a note here, because we are not locking into a version, this could change and cause unwanted issues.